build-tools/packages/build-cli/src/commands/merge/branches.ts

```diff
@@ -138,11 +138,15 @@ export default class MergeBranch extends BaseCommand<typeof MergeBranch> {
 		this.branchesToCleanup.push(tempBranchToCheckConflicts);
 
 		await this.gitRepo.gitClient
-			.checkoutBranch(tempBranchToCheckConflicts, flags.target)
-			.branch(["--set-upstream-to", `${this.remote}/${tempBranchToCheckConflicts}`]);
+			.checkoutBranch(
+				tempBranchToCheckConflicts,
+				`refs/remotes/${this.remote}/${flags.target}`,
+			)
+			.branch(["--set-upstream-to", `${tempBranchToCheckConflicts}`]);
```

Comment:
This doesn't seem right. This is creating a branch foo-automation at refs/remotes/${this.remote}/${flags.target}, but then setting the upstream of that branch to foo-automation. Maybe you meant this?

```suggestion
			.checkoutBranch(
				tempBranchToCheckConflicts,
				`refs/remotes/${this.remote}/${flags.target}`,
			)
			.branch(["--set-upstream-to", `${this.remote}/${tempBranchToCheckConflicts}`]);
```

Reply:
But why does an upstream branch have to be set at all? This is just a temporary branch used locally, right?

Reply:
yeah. I should not set temporary branch to upstream.